### ISSUE TITLE
Can't Disable Integration

### ISSUE THREAD

**Issue by corneliusio** [verbb/formie#113]
**Description**
Cannot disable any integrations in the main Formie settings.
As a side-note, was also unable to delete one integration until first deleting a different one—not sure how to reproduce though.

**Steps to reproduce**
1. Add integration at /admin/formie/settings and save.
2. Switch integration to disabled and save.

### Form settings
No Form

**Additional info**
- Plugin version: 1.2.14
- Craft version: 3.5.11.1
- Multi-site: No

**Additional context**
```
PDOException: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'enabled' cannot be null in /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/Command.php:1293
Stack trace:
#0 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/Command.php(1293): PDOStatement->execute()
#1 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/Command.php(1093): yii\db\Command->internalExecute('UPDATE `formie_...')
#2 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/ActiveRecord.php(332): yii\db\Command->execute()
#3 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/BaseActiveRecord.php(810): yii\db\ActiveRecord::updateAll(Array, Array)
#4 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/ActiveRecord.php(676): yii\db\BaseActiveRecord->updateInternal(NULL)
#5 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/BaseActiveRecord.php(681): yii\db\ActiveRecord->update(false, NULL)
#6 /Users/cornelius/Git/path/to/project/vendor/verbb/formie/src/services/Integrations.php(382): yii\db\BaseActiveRecord->save(false)
#7 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/services/ProjectConfig.php(1124): verbb\formie\services\Integrations->handleChangedIntegration(Object(craft\events\ConfigEvent))
#8 [internal function]: craft\services\ProjectConfig->handleChangeEvent(Object(craft\events\ConfigEvent))
#9 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/base/Component.php(627): call_user_func(Array, Object(craft\events\ConfigEvent))
#10 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/services/ProjectConfig.php(696): yii\base\Component->trigger('updateItem', Object(craft\events\ConfigEvent))
#11 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/services/ProjectConfig.php(487): craft\services\ProjectConfig->_processConfigChangesInternal('formie.integrat...', true, 'Save the \xE2\x80\x9Cmai...')
#12 /Users/cornelius/Git/path/to/project/vendor/verbb/formie/src/services/Integrations.php(338): craft\services\ProjectConfig->set('formie.integrat...', Array, 'Save the \xE2\x80\x9Cmai...')
#13 /Users/cornelius/Git/path/to/project/vendor/verbb/formie/src/controllers/IntegrationsController.php(82): verbb\formie\services\Integrations->saveIntegration(Object(verbb\formie\integrations\emailmarketing\Mailchimp))
#14 [internal function]: verbb\formie\controllers\IntegrationsController->actionSaveIntegration()
#15 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#16 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/base/Controller.php(180): yii\base\InlineAction->runWithParams(Array)
#17 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/web/Controller.php(189): yii\base\Controller->runAction('save-integratio...', Array)
#18 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('save-integratio...', Array)
#19 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/web/Application.php(285): yii\base\Module->runAction('formie/integrat...', Array)
#20 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/web/Application.php(588): craft\web\Application->runAction('formie/integrat...', Array)
#21 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/web/Application.php(264): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#22 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#23 /Users/cornelius/Git/path/to/project/web/index.php(28): yii\base\Application->run()
#24 /Users/cornelius/.composer/vendor/laravel/valet/server.php(191): require('/Users/corneliu...')
#25 {main}

Next yii\db\IntegrityException: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'enabled' cannot be null
The SQL being executed was: UPDATE `formie_integrations` SET `enabled`=NULL, `dateUpdated`='2020-09-21 00:57:02' WHERE `id`=8 in /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/Schema.php:677
Stack trace:
#0 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/Command.php(1298): yii\db\Schema->convertException(Object(PDOException), 'UPDATE `formie_...')
#1 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/Command.php(1093): yii\db\Command->internalExecute('UPDATE `formie_...')
#2 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/ActiveRecord.php(332): yii\db\Command->execute()
#3 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/BaseActiveRecord.php(810): yii\db\ActiveRecord::updateAll(Array, Array)
#4 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/ActiveRecord.php(676): yii\db\BaseActiveRecord->updateInternal(NULL)
#5 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/db/BaseActiveRecord.php(681): yii\db\ActiveRecord->update(false, NULL)
#6 /Users/cornelius/Git/path/to/project/vendor/verbb/formie/src/services/Integrations.php(382): yii\db\BaseActiveRecord->save(false)
#7 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/services/ProjectConfig.php(1124): verbb\formie\services\Integrations->handleChangedIntegration(Object(craft\events\ConfigEvent))
#8 [internal function]: craft\services\ProjectConfig->handleChangeEvent(Object(craft\events\ConfigEvent))
#9 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/base/Component.php(627): call_user_func(Array, Object(craft\events\ConfigEvent))
#10 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/services/ProjectConfig.php(696): yii\base\Component->trigger('updateItem', Object(craft\events\ConfigEvent))
#11 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/services/ProjectConfig.php(487): craft\services\ProjectConfig->_processConfigChangesInternal('formie.integrat...', true, 'Save the \xE2\x80\x9Cmai...')
#12 /Users/cornelius/Git/path/to/project/vendor/verbb/formie/src/services/Integrations.php(338): craft\services\ProjectConfig->set('formie.integrat...', Array, 'Save the \xE2\x80\x9Cmai...')
#13 /Users/cornelius/Git/path/to/project/vendor/verbb/formie/src/controllers/IntegrationsController.php(82): verbb\formie\services\Integrations->saveIntegration(Object(verbb\formie\integrations\emailmarketing\Mailchimp))
#14 [internal function]: verbb\formie\controllers\IntegrationsController->actionSaveIntegration()
#15 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#16 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/base/Controller.php(180): yii\base\InlineAction->runWithParams(Array)
#17 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/web/Controller.php(189): yii\base\Controller->runAction('save-integratio...', Array)
#18 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('save-integratio...', Array)
#19 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/web/Application.php(285): yii\base\Module->runAction('formie/integrat...', Array)
#20 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/web/Application.php(588): craft\web\Application->runAction('formie/integrat...', Array)
#21 /Users/cornelius/Git/path/to/project/vendor/craftcms/cms/src/web/Application.php(264): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#22 /Users/cornelius/Git/path/to/project/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#23 /Users/cornelius/Git/path/to/project/web/index.php(28): yii\base\Application->run()
#24 /Users/cornelius/.composer/vendor/laravel/valet/server.php(191): require('/Users/corneliu...')
#25 {main}
Additional Information:
Array
(
    [0] => 23000
    [1] => 1048
    [2] => Column 'enabled' cannot be null
)
```

**Comment by engram-design** [comment]
That's odd - I enable/disable integrations all the time! Are you using MySQL or Postgres?

**Comment by corneliusio** [comment]
MySQL

**Comment by engram-design** [comment]
Thanks, and your version?

**Comment by corneliusio** [comment]
5.7.31

**Comment by engram-design** [comment]
Ah, I see the issue, down to my local install. Fixed in [1.2.15](https://github.com/verbb/formie/releases/tag/1.2.15).

I wasn't able to reproduce the deleting integration issue, but if it happens again, let me know if you get an error, or shoot through your log files.

**Comment by corneliusio** [comment]
Fantastic, thanks for the quick fix!
